core/tee/tee_obj.c

```diff
@@ -53,6 +53,25 @@ TEE_Result tee_obj_get(struct tee_ta_ctx *ctx, uint32_t obj_id,
 	return TEE_ERROR_BAD_PARAMETERS;
 }
 
+TEE_Result tee_obj_get1(struct tee_ta_ctx *ctx, uint32_t obj_id,
+		       struct tee_obj **obj)
+{
+	struct tee_obj *o;
+
+	TAILQ_FOREACH(o, &ctx->objects, link) {
+		if (obj_id == (uint32_t) o) {
+			if (o->info.objectType == TEE_TYPE_CORRUPTED_OBJECT) {
+				tee_obj_close(ctx, o);
```

Comment:
This close could be done in libutee instead, the rationale for that would be to only have kernel space doing what user space can't do.